flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandlerTest.java

```diff
@@ -167,23 +164,49 @@ public void testUnrecoverableErrorCheck() {
 			new Exception(new SuppressRestartsException(new Exception()))));
 	}
 
+	@Test
+	public void testGlobalFailureHandling() {
+		final FailureHandlingResult result = executionFailureHandler.getGlobalFailureHandlingResult(
+			new Exception("test failure"));
+
+		assertTrue(result.canRestart());
+		assertEquals(restartDelayMs, result.getRestartDelayMS());
+		assertEquals(
+			StreamSupport.stream(failoverTopology.getFailoverVertices().spliterator(), false)
+			.map(FailoverVertex::getExecutionVertexID)
+			.collect(Collectors.toSet()),
+			result.getVerticesToRestart());
+		try {
```

Comment:
I think it's ok not to test here that `result.getError()` throws if `result.canRestart()` returns true. It's covered by other tests. Imo there should be only one unit test asserting this behavior in `FailureHandlingResultTest`.

Reply:
Ok. Removed the unnecessary checks from this case.